code/game/objects/items/trash.dm

```diff
@@ -1,12 +1,761 @@
+	. = ..()
+	forceMove(get_turf(hit_atom))
+	update_icon(UPDATE_ICON)
+
+/obj/effect/decal/cleanable/garbage/can_bloodcrawl_in()
+	return FALSE
+
+/obj/effect/decal/cleanable/garbage/Initialize(mapload)
+	. = ..()
+	update_icon(UPDATE_ICON)
+	if(!.)
+		dry_timer = addtimer(CALLBACK(src, PROC_REF(dry)),  8 MINUTES, TIMER_STOPPABLE)
+
+	if(prob(10))
+		overlays += image('icons/obj/trash.dmi', "big_stinkum")
+
+/obj/effect/decal/cleanable/garbage/update_icon()
+	var/turf/T = get_turf(src)
+	if(T && (T.opacity && T.density || locate(/obj/structure/window) in T))
+		plane = GAME_PLANE
+		layer = ABOVE_WINDOW_LAYER
+		icon_state = "splatter[pick("1","2","3","4","5","6")]"
+	color = basecolor
+	..()
+
+/obj/effect/decal/cleanable/garbage/proc/dry()
+	name = dryname
+	desc = drydesc
+	basecolor = "#5f5700"
+	color = basecolor
+	amount = 0
+	alpha = 255
+
+/obj/effect/decal/cleanable/garbage/drip
+	name = "drips of liquid garbage"
+	desc = "A drop of mess in an ocean of clean."
+	icon = 'icons/effects/drip.dmi'
+	icon_state = "1"
+	random_icon_states = list("1", "2", "3", "4", "5")
+	amount = 0
+	bloodiness = 0
+	mergeable_decal = FALSE
+
+/obj/effect/decal/cleanable/garbage/drip/Initialize(mapload)
+	. = ..()
+	pixel_x = rand(-6, 6)
+	pixel_y = rand(-6, 6)
+
+/obj/effect/decal/cleanable/garbage/gibs
+	//colored and using custom sprites for each garbage type
+	name = "congealed blob of liquid garbage"
+	desc = "rancid, putrid, fetid, grossness, just looking at it makes you sick."
+	random_icon_states = list("gib1", "gib2", "gib4", "gib5", "gib6", "gibmid1","gibmid2","gibmid3","gibtorso_flesh")
+	plane = GAME_PLANE
+	layer = LOW_ITEM_LAYER
+
+/obj/effect/decal/cleanable/garbage/gibs/dry()
+	return
+
+/obj/effect/decal/cleanable/garbage/gibs/Crossed(atom/movable/O)
+	. = ..()
+	if(!isliving(O))
+		return
+
+	var/mob/living/M = O
+	if(M.mob_size <= MOB_SIZE_SMALL)
+		return
+
+	//dirty shoes
+	playsound(loc,'sound/items/garbage_tread.ogg', 50, 1, -1)
+	new /obj/effect/decal/cleanable/garbage(get_turf(src))
+	qdel(src)
+
+/obj/effect/decal/cleanable/garbage/gibs/Moved(atom/OldLoc, Dir, Forced)
+	. = ..()
+
+	if(!isturf(loc))
+		return
+
+	new /obj/effect/decal/cleanable/garbage(loc)
+
+
+
+
+
+
+//[x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x]
+// ---------- TRASH DECAYING PROC-----------------------------------------------------
+//[x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x][x]
+
 //Items labled as 'trash' for the trash bag.
 //TODO: Make this an item var or something...
 
+//decay timings 1:2min --> 10 minutes, 20 minutes, 40 minutes
+#define DECAY_TIMINGS list(5, 10, 20, INFINITY)
+
+/datum/trash_handler
+//this is so non-obj/item/trash things can also rot by calling this proc while having correct variables
+/datum/trash_handler/proc/handle_decay(var/obj/item/trash/input_item, timing_multiplier=1, do_icon_operations=TRUE)
```

Comment:
Doing this every process tick for a ton of items might cause performance issues. Or will at least cost quite a bit. It might be better to make a subsystem for this which has a queue to handle decay.
Since the decay time is always the same you can use this to optimise the handling of it.

pseudo code:
```dm
/item/start_decay()
 SStrash.add_decaying_item(src)

/SStrash
 var/list/queue = list()


SStrash/add_decaying_item(item)
 queue[item] = world.time + 2 MINUTES

SStrash/fire()
 for(var/item in queue)
  if(queue[item] < world.time)
   break
  handle_decay(item)
  queue -= item // or some other cutting mechanism
```

```